include/commissioner/commissioner.hpp

```diff
@@ -1133,6 +1133,15 @@ class Commissioner
      * @param[in]      aJoinerId      A Joiner ID.
      */
     static void AddJoiner(ByteArray &aSteeringData, const ByteArray &aJoinerId);
+
+    /**
+     * @brief Return the commissioner version.
+     *
+     * @return A version string in format of <MAJOR>.<MINOR>.<PATCH>[-<GIT_REVISION>].
+     *         The GIT_REVISION is included only when the commissioner is built in a
+     *         git repository.
+     */
```

Comment:
Better not include git revision to make the format consistent? 
Maybe we can add another api to get the git revision if necessary.

Reply:
Actually, I want the git revision more than the semantic version numbers. It is the usual situation that we want the exact commit ID when there are IOP issues. I believe OpenThread and ot-br-posix are already including the git revision in the version string and we are just following it here.

What is different here is that we provide an API for getting the version string but not defining a macro or use auto-generated version file. This is because ot-commissioner can be built and installed to system directory and used in a project that is not using CMake or automake tools.

Reply:
I rechecked https://semver.org/#spec-item-10. 

Seems we need to use the format:
`<MAJOR>.<MINOR>.<PATCH>[+<GIT_REVISION>]` because `GIT_REVISION` should be `Build metadata`.